src/main/java/bio/terra/dao/DataProjectDao.java

```diff
@@ -0,0 +1,118 @@
+package bio.terra.dao;
+
+import bio.terra.configuration.DataRepoJdbcConfiguration;
+import bio.terra.dao.exception.DataProjectNotFoundException;
+import bio.terra.metadata.DatasetDataProjectSummary;
+import bio.terra.metadata.StudyDataProjectSummary;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.UUID;
+
+@Repository
+public class DataProjectDao {
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public DataProjectDao(DataRepoJdbcConfiguration jdbcConfiguration) {
+        jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+    }
+
+    public UUID createStudyDataProject(StudyDataProjectSummary studyDataProjectSummary) {
+        String sql = "INSERT INTO study_data_project (study_id, project_resource_id) VALUES " +
+            "(:study_id, :project_resource_id)";
+        MapSqlParameterSource params = new MapSqlParameterSource()
+            .addValue("study_id", studyDataProjectSummary.getStudyId())
+            .addValue("project_resource_id", studyDataProjectSummary.getProjectResourceId());
+        DaoKeyHolder keyHolder = new DaoKeyHolder();
+        jdbcTemplate.update(sql, params, keyHolder);
+        return keyHolder.getId();
+    }
+
+    private StudyDataProjectSummary retrieveStudyDataProjectByColumn(String column, UUID value) {
+        try {
+            String sql = String.format("SELECT * FROM study_data_project WHERE %s = :%s", column, column);
```

Comment:
This looks dodgy to me. I don't think we should use `*` as the select list. Not so sure about passing in a column name string and injecting that into a SQL statement.  Aren't there only two columns?

Reply:
fixing like we discussed.